patroni/postgresql/slots.py

```diff
@@ -331,11 +332,10 @@ def _drop_incorrect_slots(self, cluster: Cluster, slots: Dict[str, Any], paused:
 
         :param cluster: cluster state information object.
         :param slots: dictionary of desired slot names as keys with slot attributes as a dictionary value, if known.
-        :param paused: ``True`` if the patroni cluster is currently in a paused state.
         """
         # drop old replication slots which are not presented in desired slots.
         for name in set(self._replication_slots) - set(slots):
-            if not paused and not self.ignore_replication_slot(cluster, name):
+            if not global_config.is_paused and not self.ignore_replication_slot(cluster, name):
```

Comment:
Ideally I would say this change should go as part of a separate commit, but maybe we could at least adjust the commit message and the PR description to say we are adjusting a few leftovers from the global config PR?

Reply:
You are right, it is an oversight of #2935, I will create a separate PR for it.